src/actions/admin.js

```diff
@@ -0,0 +1,52 @@
+import * as api from '../services/api';
+import History from '../history';
+
+import { GET_PROTECTED_DATA, MODLIST_ERROR } from './types';
+
+export const getModlist = () => {
+  return async dispatch => {
+    try {
+      const response = await api.get({ url: 'moderatorlist' });
+      dispatch({ type: GET_PROTECTED_DATA, payload: response.data });
+    } catch (err) {
+      dispatch({ type: MODLIST_ERROR, payload: err });
+      History.push('/moderatorlist');
+    }
+  };
+};
+
+export function demoteMod(id) {
+  return async dispatch => {
+    try {
+      await api.put({ url: 'demote/' + id });
+      History.push('/moderatorlist');
```

Comment:
Think it'd be a good idea to have the url imported as a constant, or have a convenience method for pushing `/moderatorlist`. This way, if the url changes for what ever reason, it's much easier to change it project wide.

Applicable to other similar cases as well.